numpy/core/_dtype_ctypes.py

```diff
@@ -33,17 +33,28 @@ def _from_ctypes_array(t):
 
 
 def _from_ctypes_structure(t):
-    # TODO: gh-10533, gh-10532
-    fields = []
-    for item in t._fields_:
-        if len(item) > 2:
-            raise TypeError(
+    # TODO: gh-10533
+    if hasattr(t,"_pack_"):
+        fields = {}
+        for item in t._fields_:
+            if len(item) > 2:
+                raise TypeError(
                 "ctypes bitfields have no dtype equivalent")
-        fname, ftyp = item
-        fields.append((fname, dtype_from_ctypes_type(ftyp)))
+            fname, ftyp = item
+            fields[fname] = dtype_from_ctypes_type(ftyp), t._pack_
```

Comment:
This looks wrong to me - you need to be keeping track of the size so far, and then rounding up to the next multiple of `_pack_` at each step.

Reply:
I am not sure I understood the formula you provided for offset. This is how I would calculate it: 
`offset = (((offset + _pack_ - 1) // _pack_) + 1 ) * _pack_` - the first offset is set to 0 and this value is calculated after an item is added (as I need it's size).

Reply:
Sorry, I meant that as two steps:
```
offset += field.itemsize
offset = (offset + _pack_ - 1) // _pack_ * _pack_)
```